crates/shared/src/signature_validator.rs

```diff
@@ -58,71 +47,6 @@ pub trait SignatureValidating: Send + Sync {
     ) -> Result<u64, SignatureValidationError>;
 }
 
-pub struct Web3SignatureValidator {
```

Comment:
Moved almost verbatim to `crates/shared/src/signature_validator/web3.rs`, changes:
- Added warning logs for validating signatures where pre-interactions are specified.
- Removed the `SignatureValidating::validate_signature` trait method implementation, as it was removed from the trait.